tests/operators/test_agnostic_aggregate_check.py

```diff
@@ -92,194 +105,188 @@ def setUp(self):
-            bigquery_conn_id="bigquery", use_legacy_sql=False, gcp_conn_id="bigquery"
-        )
-        df = hook.get_pandas_df(
-            sql=f"SELECT * FROM {DEFAULT_SCHEMA}.aggregate_check_test"
-        )
-        assert df.count()[0] == 4
-        try:
-            a = aql.aggregate_check(
-                table=self.aggregate_table_bigquery,
+        @aql.transform
+        def get_table(input_table: Table):
+            return "SELECT * FROM {{input_table}}"
+
+        dag = get_dag()
+        with dag:
+            aggregate_table_bigquery = get_table(self.aggregate_table_bigquery)
+            aql.aggregate_check(
+                table=aggregate_table_bigquery,
                 check="select count(*) FROM {{table}}",
                 greater_than=4,
                 less_than=4,
             )
-            a.execute({"run_id": "foo"})
-            assert True
-        except ValueError:
-            assert False
+        run_dag(dag)
 
     def test_exact_value_sqlite(self):
-        hook = SqliteHook(sqlite_conn_id="sqlite_conn")
-        df = hook.get_pandas_df(sql="SELECT * FROM aggregate_check_test")
-        assert df.count()[0] == 4
-        try:
-            a = aql.aggregate_check(
-                table=self.aggregate_table_sqlite,
+        @aql.transform
+        def get_table(input_table: Table):
+            return "SELECT * FROM {{input_table}}"
+
+        dag = get_dag()
+        with dag:
+            aggregate_table_sqlite = get_table(self.aggregate_table_sqlite)
+            aql.aggregate_check(
+                table=aggregate_table_sqlite,
                 check="select count(*) FROM {{table}}",
                 greater_than=4,
                 less_than=4,
             )
-            a.execute({"run_id": "foo"})
-            assert True
-        except ValueError:
-            assert False
+        run_dag(dag)
 
     def test_range_values(self):
-        hook = PostgresHook(schema="pagila", postgres_conn_id="postgres_conn")
-        df = hook.get_pandas_df(
-            sql="SELECT * FROM airflow_test_dag.aggregate_check_test"
+        aggregate_table = Table(
+            "aggregate_check_test",
+            database="pagila",
+            conn_id="postgres_conn",
+            schema="airflow_test_dag",
         )
-        assert df.count()[0] == 4
-        try:
-            a = aql.aggregate_check(
-                table=self.aggregate_table,
+
+        @aql.transform
+        def get_table(input_table: Table):
+            return "SELECT * FROM {{input_table}}"
+
+        dag = get_dag()
+        with dag:
+            table = get_table(aggregate_table)
+            aql.aggregate_check(
+                table=table,
                 check="select count(*) FROM {{table}}",
                 greater_than=2,
                 less_than=6,
             )
-            a.execute({"run_id": "foo"})
-            assert True
-        except ValueError:
-            assert False
+        run_dag(dag)
```

Comment:
I'm noticing that a LOT of this code is being repeated, and I think we can cut down on a lot of the code using [pytest parameterize](https://docs.pytest.org/en/6.2.x/parametrize.html)

I think the two main cases here are 1) DAGs that pass and 2) DAGs that fail. You can parameterize the parameters like we have [here](https://github.com/astro-projects/astro/blob/main/tests/operators/test_agnostic_merge.py#L48-L80). This will also make it really easy for us to run these tests against all databases.

Reply:
Done